llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -5091,8 +5101,18 @@ bool SelectionDAG::canCreateUndefOrPoison(SDValue Op, const APInt &DemandedElts,
   case ISD::BITCAST:
   case ISD::BUILD_VECTOR:
   case ISD::BUILD_PAIR:
+  case ISD::SELECT:
     return false;
 
+  case ISD::SELECT_CC: {
+    // Integer select_cc cannot create undef or poison.
+    if (Op.getOperand(0).getValueType().isInteger())
+      return false;
+
+    // FIXME: Handle FP compares.
```

Comment:
Can this be combined with the following SETCC case, which has FP handling?

Reply:
Hmm, yes, it is probably just as simple as that (if just making sure we get the cond code  from correct operand). I will updated the patch.